csrc/device_lower/analysis/predicate_elimination.cpp

```diff
@@ -201,12 +231,19 @@ class ProducerConsumerPairAnalyzer : public OptOutDispatch {
 
  private:
   ProducerConsumerPairAnalyzer(
-      const std::unordered_map<IterDomain*, IterDomain*>& c2p)
-      : c2p_(c2p) {}
+      const std::unordered_map<IterDomain*, IterDomain*>& c2p,
+      const std::unordered_set<IterDomain*> index_ids)
+      : c2p_(c2p), index_ids_(index_ids) {}
 
   // Returns true if no out-of-bound accesses could occur with a
   // producer
   bool needsPredicate(IterDomain* consumer_id) {
+    // TODO: check that this consumer_id is actually involved in indexing the
+    // producer. If it is not connected to the producer allocation domain in
+    // the broadcast graph, then we can skip processing it.
+    if (index_ids_.find(consumer_id) == index_ids_.end()) {
+      return false;
```

Comment:
Is there any existing test case or benchmark that hits this path?

Reply:
I don't think any tests other than the one introduced in this PR will hit this path, but to be sure I'm going to introduce an `NVF_THROW` here and run the CI just to check.

Reply:
We hit this path whenever we have a `BroadcastOp`. All of the new Broadcast domains in the consumer root domain normally pass through here but are returned immediately below at line 260.